File: Sources/YToolsCore/ClipboardTextPolicy.swift
import Foundation

public struct ClipboardTextPolicy: Equatable, Sendable {
    public let maximumCharacters: Int

    public init(maximumCharacters: Int) {
        self.maximumCharacters = max(1, maximumCharacters)
    }

    public func shouldStore(_ text: String) -> Bool {
        !text.isEmpty && text.count <= maximumCharacters
    }
}

File: src/YTools.Windows/Core/ClipboardTextPolicy.cs
namespace YTools.Core;

public readonly struct ClipboardTextPolicy
{
    public ClipboardTextPolicy(int maximumCharacters)
    {
        MaximumCharacters = Math.Max(1, maximumCharacters);
    }

    public int MaximumCharacters { get; }

    public bool ShouldStore(string text)
    {
        return !string.IsNullOrEmpty(text) && text.Length <= MaximumCharacters;
    }
}
